homeassistant/components/recorder/core.py

```diff
@@ -586,7 +586,8 @@ def _setup_recorder(self) -> None | int:
         while tries <= self.db_max_retries:
             try:
                 self._setup_connection()
-                return migration.get_schema_version(self)
+                assert self.get_session is not None
```

Comment:
Instead of asserting this everywhere, just working against the typing, maybe it would be a better option to type `get_session` as non-optional, like this: https://github.com/home-assistant/core/blob/4408ad82eb26f0b64b775d9ab60c00fd53820067/homeassistant/core.py#L239

Reply:
Or don't assign it `None` in the first place, but just declare it.

Reply:
Removed the asserts and added a better exception when its called and the database isn't connected.